src/main/java/com/j256/ormlite/field/types/OffsetTimeCompatType.java

```diff
@@ -0,0 +1,54 @@
+package com.j256.ormlite.field.types;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.format.DateTimeFormatter;
+
+import com.j256.ormlite.field.FieldType;
+import com.j256.ormlite.field.SqlType;
+import com.j256.ormlite.misc.SqlExceptionUtil;
+import com.j256.ormlite.support.DatabaseResults;
+
+/**
+ * A custom persister that is able to store the java.time.OffsetTime class in the database as Timestamp With Time Zone object.
+ * This class should be used only when database used does not support Time With Time Zone, since it converts java.time.OffsetTime
+ * to java.time.OffsetDateTime, fixing date part at epoch, to be stored as Timestamp With Time Zone.
+ *
+ * @author graynk
+ */
+public class OffsetTimeCompatType extends OffsetTimeType {
+
+    private static final OffsetTimeCompatType singleton = isJavaTimeSupported() ? new OffsetTimeCompatType() : null;
+    public static OffsetTimeCompatType getSingleton() { return singleton; }
+    private OffsetTimeCompatType() { super(SqlType.OFFSET_DATE_TIME); }
+    protected OffsetTimeCompatType(SqlType sqlType, Class<?>[] classes) { super(sqlType, classes); }
+
+    @Override
+    public Object parseDefaultString(FieldType fieldType, String defaultStr) throws SQLException {
+        try {
+            return OffsetDateTime.parse(defaultStr, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss[.SSS]x"));
```

Comment:
I've been building tests over on the JDBC side so sorry for not catching everything at once.

This here is going to cause problems as it is incompatible with regular `OffsetTimeType`.

If we did `@DatabaseField(defaultValue = "01:23:45+0100")` on `OffsetTime`, it would work everywhere except Postgres or H2 which uses the compat persister instead.

A better solution here would be to make the date part optional, parse it into an `OffsetTime` and then convert to `OffsetDateTime`.

Example: https://github.com/Bo98/ormlite-core/commit/bd8ee75dbe5d63b3027fa3e99006bea83dada210 (not tested everywhere but seems to work)

Reply:
I believe I've integrated all of your fixes now

Reply:
I'll have a look come the end of the week when I'm back home.